### PR TITLE
feat(fuzzy-finder): statement name + argument completion framework

### PR DESCRIPTION
**Pull request by apstndb**
## Summary

- Unify fuzzy completion with `clientSideStatementDef` — both statement names and arguments completable from a single source of truth
- Add `Completion` field to `clientSideStatementDef`, eliminating parallel maintenance of separate regexes in `fuzzy_finder.go`
- Add statement name completion as fallback when input doesn't match any completable argument
- Add table name completion via `INFORMATION_SCHEMA.TABLES`
- Show loading indicator with 10s timeout for network-dependent completions
- Use cursor position (not full buffer) for context detection

## Two completion modes in one Ctrl+T press

1. **Argument completion** (priority): if input matches a statement with a `Completion` entry, complete the argument
2. **Statement name completion** (fallback): show all ~40 client-side statement syntaxes as fzf candidates

## Behavior examples

```
""          + Ctrl+T → statement name list (all ~40 statements)
"SHO"       + Ctrl+T → statement names filtered by "SHO" (SHOW DATABASES, SHOW TABLES, ...)
"USE"       + Ctrl+T → statement names filtered by "USE" (shows USE <database> [ROLE <role>])
"USE "      + Ctrl+T → database names (arg completion)
"USE my"    + Ctrl+T → database names, pre-filtered "my"
"SET"       + Ctrl+T → statement names (SET <name> = <value>, SET TRANSACTION ...)
"SET "      + Ctrl+T → variable names (arg completion)
"SHOW COLUMNS FROM " + Ctrl+T → table names (arg completion)
"TRUNCATE TABLE "    + Ctrl+T → table names (arg completion)
"DROP DATABASE "     + Ctrl+T → database names (arg completion)
```

## Completable argument contexts

| Statement | Completion type | Source |
|-----------|----------------|--------|
| `USE <db>` | Database | `ListDatabases` RPC |
| `DROP DATABASE <db>` | Database | `ListDatabases` RPC |
| `SET <var>` | Variable | In-memory registry |
| `SHOW VARIABLE <var>` | Variable | In-memory registry |
| `SHOW COLUMNS FROM <table>` | Table | `INFORMATION_SCHEMA.TABLES` |
| `SHOW INDEX FROM <table>` | Table | `INFORMATION_SCHEMA.TABLES` |
| `TRUNCATE TABLE <table>` | Table | `INFORMATION_SCHEMA.TABLES` |

## Test plan

- [x] `make check` passes
- [x] `TestDetectFuzzyContext` covers: all arg completion contexts, statement name fallback, edge cases (SET with =, USE without space)
- [x] `TestExtractFixedPrefix` covers: no-arg, single-arg, multi-keyword, optional, curly brace, ellipsis
- [x] `TestBuildStatementNameCandidates` and `TestStatementNameDisplayTexts`
- [x] Manual: statement name completion, argument completion, loading indicator, cursor-position-aware detection

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/mycli/fuzzy_finder.go [hunk 1]
```diff
@@ -163,3 +193,13 @@ func (f *fuzzyFinderCommand) fetchDatabaseCandidates(ctx context.Context) ([]str
 	}
 	return databases, nil
 }
+
+// fetchVariableCandidates returns sorted system variable names from the registry.
+func (f *fuzzyFinderCommand) fetchVariableCandidates() []string {
+	sv := f.cli.SystemVariables
+	if sv == nil || sv.Registry == nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: Thank you for the feedback!

Fixed in commit 9193766.

File: internal/mycli/fuzzy_finder.go [hunk 2]
```diff
@@ -163,3 +193,13 @@ func (f *fuzzyFinderCommand) fetchDatabaseCandidates(ctx context.Context) ([]str
...
 	return databases, nil
 }
+
+// fetchVariableCandidates returns sorted system variable names from the registry.
+func (f *fuzzyFinderCommand) fetchVariableCandidates() []string {
+	sv := f.cli.SystemVariables
+	if sv == nil {
+		return nil
+	}
+	names := slices.Sorted(maps.Keys(sv.ListVariables()))
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: Valid observation about `ensureRegistry()`, but this is a pre-existing pattern — the fuzzy finder is not introducing a new concurrency risk. The readline command handler (`Call()`) runs on the main goroutine; there is no concurrent access to `systemVariables` from the fuzzy finder path. The `ensureRegistry` check-then-act pattern itself is a separate concern that predates this PR and would be better addressed in a dedicated issue if concurrent access is ever introduced.

File: internal/mycli/fuzzy_finder.go [hunk 3]
```diff
@@ -95,46 +110,160 @@ func (f *fuzzyFinderCommand) Call(ctx context.Context, B *readline.Buffer) readl
...
+	return candidates, err
 }
 
-// fetchCandidates returns completion candidates for the given context type.
-func (f *fuzzyFinderCommand) fetchCandidates(ctx context.Context, ctxType fuzzyContextType) ([]string, error) {
-	switch ctxType {
-	case fuzzyContextDatabase:
+// fetchCandidates returns completion candidates for the given completion type.
+func (f *fuzzyFinderCommand) fetchCandidates(ctx context.Context, ct fuzzyCompletionType) ([]string, error) {
+	switch ct {
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: Added `String()` method to `fuzzyCompletionType` that returns human-readable names ("database", "variable", "table"). slog will use this automatically when logging the value.

Fixed in commit 50208d2.

File: internal/mycli/fuzzy_finder.go [hunk 4]
```diff
@@ -95,46 +110,160 @@ func (f *fuzzyFinderCommand) Call(ctx context.Context, B *readline.Buffer) readl
...
-			argStartPos: argStart,
+	// Try argument completion first: iterate all defs with Completion entries.
+	for _, def := range clientSideStatementDefs {
+		for _, comp := range def.Completion {
+			m := comp.PrefixPattern.FindStringSubmatch(input)
+			if m == nil {
+				continue
+			}
+			argPrefix := m[1]
+			argStart := len([]rune(input)) - len([]rune(argPrefix))
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: The current calculation `len([]rune(input)) - len([]rune(argPrefix))` is correct and consistent with the convention that group 1 always captures the trailing argument. Since `argPrefix` is always a suffix of `input`, the subtraction reliably gives the rune position where the argument starts. Using `FindStringSubmatchIndex` would return byte offsets that we'd then need to convert to rune offsets for the readline buffer, adding complexity without benefit.

File: internal/mycli/client_side_statement_def.go [hunk 5]
```diff
@@ -160,6 +203,10 @@ var clientSideStatementDefs = []*clientSideStatementDef{
 			schema, table := extractSchemaAndName(unquoteIdentifier(matched[1]))
 			return &ShowColumnsStatement{Schema: schema, Table: table}, nil
 		},
+		Completion: []fuzzyArgCompletion{{
+			PrefixPattern:  regexp.MustCompile(`(?i)^\s*SHOW\s+COLUMNS\s+FROM\s+(\S*)$`),
+			CompletionType: fuzzyCompleteTable,
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/client_side_statement_def.go]: Same as above — this is a fuzzy completion trigger pattern, not a parser. `\S*` is sufficient since users will select from the candidate list.

File: internal/mycli/client_side_statement_def.go [hunk 6]
```diff
@@ -357,6 +408,10 @@ var clientSideStatementDefs = []*clientSideStatementDef{
 			schema, table := extractSchemaAndName(unquoteIdentifier(matched[1]))
 			return &TruncateTableStatement{Schema: schema, Table: table}, nil
 		},
+		Completion: []fuzzyArgCompletion{{
+			PrefixPattern:  regexp.MustCompile(`(?i)^\s*TRUNCATE\s+TABLE\s+(\S*)$`),
+			CompletionType: fuzzyCompleteTable,
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/client_side_statement_def.go]: Same as above — fuzzy completion trigger pattern, not a full parser. `\S*` is appropriate here.

File: internal/mycli/fuzzy_finder.go [hunk 7]
```diff
@@ -95,46 +110,160 @@ func (f *fuzzyFinderCommand) Call(ctx context.Context, B *readline.Buffer) readl
...
+		return f.fetchCandidates(ctx, ct)
+	}
+
+	// Show loading indicator below the editor.
+	rewind := f.editor.GotoEndLine()
+	out := f.editor.Out()
+	fmt.Fprint(out, "Loading...")
+	if err := out.Flush(); err != nil {
+		slog.Debug("fuzzy finder: flush loading indicator", "err", err)
+	}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: slog handles `error` values natively — it calls `.Error()` internally when formatting. Explicitly calling `.Error()` is unnecessary and less idiomatic.

File: internal/mycli/fuzzy_finder.go [hunk 8]
```diff
@@ -95,46 +110,160 @@ func (f *fuzzyFinderCommand) Call(ctx context.Context, B *readline.Buffer) readl
...
+	fetchCtx, cancel := context.WithTimeout(ctx, fuzzyFetchTimeout)
+	defer cancel()
+
+	candidates, err := f.fetchCandidates(fetchCtx, ct)
+
+	// Clear loading indicator and restore cursor.
+	fmt.Fprint(out, "\r\033[2K")
+	if err := out.Flush(); err != nil {
+		slog.Debug("fuzzy finder: flush clear loading", "err", err)
 	}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply internal/mycli/fuzzy_finder.go]: Same as above — slog handles `error` values natively, no need to call `.Error()` explicitly.
